bucket_map/src/bucket_storage.rs

```diff
@@ -332,21 +332,23 @@ impl BucketStorage {
         self.stats.resize_us.fetch_add(m.as_us(), Ordering::Relaxed);
     }
 
-    /// allocate a new bucket based on 'self', but copying data from 'bucket'
-    pub fn grow(
-        &self,
+    /// allocate a new bucket, copying data from 'bucket'
+    pub fn new_resized(
+        drives: &Arc<Vec<PathBuf>>,
+        max_search: MaxSearch,
         bucket: Option<&Self>,
         capacity_pow_2: u8,
         num_elems: u64,
         elem_size: u64,
+        stats: &Arc<BucketStats>,
```

Comment:
Can `drives` and `stats` be _not_ a reference here, and have the callers call `Arc::clone()` instead? I think that's the more idiomatically Rust way.

Reply:
they could be. I'm working under the model that the function knows whether it needs to keep the reference or not. For example, if it recycles a bucket somehow, it may choose to ignore the by value arc that was passed in. So, we had an extra clone. I find it hard to make a function take a reference when it used to take a value than to go the other way.